src/lib.rs

```diff
@@ -2053,8 +2075,18 @@ impl Connection {
         // Limit output packet size to respect peer's max_packet_size limit.
         left = cmp::min(left, self.max_send_udp_payload_len());
 
-        // Limit output packet size by congestion window size.
-        left = cmp::min(left, self.recovery.cwnd_available());
+        #[allow(unused_variables)]
+        let ignore_cc = false;
+
+        #[cfg(feature = "quic-dgram")]
+        let ignore_cc = self.dgram_send_queue.has_pending() &&
```

Comment:
I am not familiar with compiler optimizations for `rustc` (rust compiler). However, assuming LLVM infrastructure behind the scene, I would say that it'd be better moving the condition expression `self.dgram_ignore_cc` as first operand because we might take advantage from branch predictor on modern CPU architecture and short circuit evaluation. To be discussed

Reply:
They should be of similar speed (Rust is based on LLVM, but in release builds does much more aggressive inlining, so `self.dgram_send_queue.has_pending()` is probably just as zero cost as `self.dgram_ignore_cc`)